packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -823,14 +827,26 @@ export interface ISharedObjectKind<TSharedObject> {
 	create(runtime: IFluidDataStoreRuntime, id?: string): TSharedObject;
 }
 
+/**
+ * Defines a kind of shared object.
+ * Used in containers to register a shared object implementation, and to create new instances of a given type of shared object.
+ * @privateRemarks
+ * Type erased reference to an {@link ISharedObjectKind} or a DataObject class.
+ * For use by the declarative API only.
+ * @public
+ */
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface SharedObjectKind<out TSharedObject = unknown>
```

Comment:
I'd recommend creating some readme.md that talks more in detail these types and intended use, and what problems they solve. Just glancing at this type (or comments above), it's not very clear on what is intended usage. Maybe we should rely on developers glancing at current usage in repo to imply these answers, but it feels like having a bit more English would be useful.

Reply:
Good point.

I have extended the doc comments here quite a bit. I picked putting the text there as its easier to find/link from the code and the readme is more public/use facing and this seems like developer facing info mostly.

If you have anything further you think I should add, let me know.